tests/Support/Constraint/StringContainsCss.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Tests\Support\Constraint;
+
+/**
+ * This constraint asserts that the string it is evaluated for contains some specific CSS, allowing for cosmetic
+ * whitespace differences.
+ *
+ * The CSS string passed in the constructor.
+ *
+ * @author Jake Hotson <jake.github@qzdesign.co.uk>
+ */
+class StringContainsCss extends CssConstraint
+{
+    /**
+     * @var string
+     */
+    private $css;
+
+    /**
+     * @param string $css
+     */
+    public function __construct(string $css)
+    {
+        parent::__construct();
+
+        $this->css = $css;
+    }
+
+    /**
+     * @return string a string representation of the constraint
+     */
+    public function toString(): string
+    {
+        return 'contains CSS `' . $this->css . '`';
+    }
+
+    /**
+     * Evaluates the constraint for the parameter `$other`.
+     *
+     * @param mixed $other value or object to evaluate
+     *
+     * @return bool `true` if the constraint is met, `false` otherwise
+     */
+    protected function matches($other): bool
+    {
+        return \preg_match(self::getCssNeedleRegularExpressionPattern($this->css), $other) > 0;
```

Comment:
The second parameter of `preg_match` needs to be a string. So we either should add a `string` type declaration to `$other` or return an early `false` if `$other` is no string.

Reply:
It's an implementation of an abstract method, so the parameter type can't be changed.  Thus I've added a check that it's a string, along with tests for that.